catalog/plugin/catalog-plugin-metacardingest-network/src/test/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeFactoryTest.java

```diff
@@ -0,0 +1,180 @@
+        Attribute attribute = attributeFactory.createAttribute(mockDescriptor, "1874xyz");
+    }
+
+    @Test
+    public void testCreateMultiValuedAttributeFromCommaSeparatedList() throws Exception {
+        when(mockDescriptor.isMultiValued()).thenReturn(true);
+        when(mockDescriptor.getName()).thenReturn(LANGUAGE);
+        when(mockType.getAttributeFormat()).thenReturn(STRING);
+
+        Attribute attribute = attributeFactory.createAttribute(mockDescriptor,
+                COMMA_SEPARATED_LANGUAGES);
+        List<Serializable> languages = attribute.getValues();
+
+        assertThat(languages, hasSize(3));
+        assertThat(languages, containsInAnyOrder(ENGLISH, FRENCH, GERMAN));
+    }
+
+    @Test
+    public void testParseAttributeValueInteger() throws Exception {
+        runParameterizedAttributeValueTest(INTEGER, "186", 186);
+    }
+
+    @Test
+    public void testParseAttributeValueDouble() throws Exception {
+        runParameterizedAttributeValueTest(DOUBLE, "0.8866432", 0.8866432);
+    }
+
+    @Test
+    public void testParseAttributeValueFloat() throws Exception {
+        runParameterizedAttributeValueTest(FLOAT, "0.123", 0.123f);
+    }
+
+    @Test
+    public void testParseAttributeValueShort() throws Exception {
+        runParameterizedAttributeValueTest(SHORT, "16", (short) 16);
+    }
+
+    @Test
+    public void testParseAttributeValueLong() throws Exception {
+        runParameterizedAttributeValueTest(LONG, "123456789", 123456789L);
+    }
+
+    @Test
+    public void testParseAttributeValueBoolean() throws Exception {
+        runParameterizedAttributeValueTest(BOOLEAN, "true", true);
+    }
+
+    @Test
+    public void testParseAttributeValueBinary() throws Exception {
+        String binaryString = "0100101010110110010010110110010";
+        runParameterizedAttributeValueTest(BINARY, binaryString, binaryString.getBytes());
+    }
+
+    @Test
+    public void testParseAttributeValueDate() throws Exception {
+        String lexicalXsdDateTime = "2001-10-26T21:32:52";
+        runParameterizedAttributeValueTest(DATE,
+                lexicalXsdDateTime,
+                DatatypeConverter.parseDateTime(lexicalXsdDateTime)
+                        .getTime());
+    }
+
+    @Test
+    public void testParseAttributeValueString() throws Exception {
+        runParameterizedAttributeValueTest(STRING, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
+    @Test
+    public void testParseAttributeValueObject() throws Exception {
+        runParameterizedAttributeValueTest(OBJECT, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
+    @Test
+    public void testParseAttributeValueGeometry() throws Exception {
+        runParameterizedAttributeValueTest(GEOMETRY, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
+    @Test
+    public void testParseAttributeValueXml() throws Exception {
+        runParameterizedAttributeValueTest(XML, SAMPLE_JSON_STRING, SAMPLE_JSON_STRING);
+    }
+
```

Comment:
If `parseAttributeValue()` remains `public`, we may want to add tests for `null` arguments and parsing errors for all argument types.